Sources/Errors.swift

```diff
@@ -4,7 +4,16 @@
 
  Custom implementations of Mappable, Convertible, or transformation functions can throw any error they desire
  */
-public struct MapperError: ErrorType {
+public enum MapperError: ErrorType {
+    case ConvertibleError(value: AnyObject?, type: Any.Type)
+    case CustomError(message: String)
```

Comment:
Maybe we should include the field here as an optional? Seems like other than the legacy case every error would have a field